src/setuptools_scm/config.py

```diff
@@ -6,7 +6,7 @@
 
 from .utils import trace
 
-DEFAULT_TAG_REGEX = r"^(?:[\w-]+-)?(?P<version>[vV]?\d+(?:\.\d+){0,2}[^\+]+)(?:\+.*)?$"
+DEFAULT_TAG_REGEX = r"^(?:[\w-]+-)?(?P<version>[vV]?\d+(?:\.\d+){0,2}[^\+]*)(?:\+.*)?$"
```

Comment:
i beleive this fix is actually correct, and by now im horrified enough of the fine details of this regex, that i want to transform it into a verbose commented regex in a future followup